pandas/core/arrays/sparse/array.py

```diff
@@ -1636,6 +1639,44 @@ def _min_max(self, kind: Literal["min", "max"], skipna: bool) -> Scalar:
         else:
             return na_value_for_dtype(self.dtype.subtype, compat=False)
 
+    def _argmin_argmax(self, kind: Literal["argmin", "argmax"]) -> int:
+
+        values = self._sparse_values
+        mask = np.asarray(isna(self))[self._sparse_index.indices]
+        func = np.argmax if kind == "argmax" else np.argmin
+
+        idx = np.arange(values.shape[0])
+        non_nans = values[~mask]
+        non_nan_idx = idx[~mask]
+
+        _candidate = non_nan_idx[func(non_nans)]
+        candidate = self._sparse_index.indices[_candidate]
```

Comment:
`.indices` is defined as a property on the `SparseIndex` class, so is not free to calculate. Better to use access once, store in local instead of accessing multiple times